src/generate-types.ts

```diff
@@ -675,6 +702,220 @@ function generateTopLevelClassBuilder(key: string, schema: oas.SchemaObject) {
   return generateTopLevelMaker(key, schema, 'build', 'ShapeOf' + oautil.typenamify(key));
 }
 
+function generateReflectionType(
+  schema: oas.SchemaObject | oas.ReferenceObject
+): ts.ObjectLiteralExpression {
+  if (oautil.isReferenceObject(schema)) {
+    const resolved = resolveRefToTypeName(schema.$ref);
+    const type: ts.Expression = resolved.qualified
+      ? ts.createPropertyAccess(resolved.qualified, 'type' + resolved.member)
+      : ts.createIdentifier('type' + resolved.member);
+    return ts.createObjectLiteral(
+      [
+        ts.createPropertyAssignment('type', ts.createStringLiteral('named')),
```

Comment:
Where are we putting the information about, for example, whether or not the property is a URL? I thought it would be under `type`?

Reply:
at the moment this does not save the format field anywhere. 

Might be usable for test data generation purposes even though afaik oats does not validate the format either so :shrug:

Reply:
I'll put the format to the type